### PR TITLE
Nh linked list pytest working

### PR DESCRIPTION
**Pull request by nxnode**

### CODE REVIEW

File: code_fellows/linked_list.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 class LinkedList:
     def __init__(self, values=None):
         self.head = None
+        self.head = None
```
Comment (W-Ely) [inline code_fellows/linked_list.py]: Looks like `self.head = None` is here 2x

File: code_fellows/linked_list.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
...
+        self.length = 0
```
Comment (W-Ely) [inline code_fellows/linked_list.py]: `self.length` isn't used, maybe it should be?  or remove it.

File: code_fellows/linked_list.py [hunk 3]
```diff
@@ -12,8 +14,40 @@ def push(self, values):
...
+    def size(self):
+        count = 0
+        currrent_node = self.head
+        while currrent_node:
+            count += 1
+            currrent_node = currrent_node.next
+        return count
+
+    def search(self, value):
+        count = 1
```
Comment (W-Ely) [inline code_fellows/linked_list.py:25]: `count = 1` the count is no longer needed in this function.

File: tests/code_fellows/test_linked_list.py [hunk 4]
```diff
@@ -1,21 +1,45 @@
...
 
 
 def test_linked_list():
-    assert LinkedList(["head"]).head.value == "head"
-    assert (
-        LinkedList([3, "three"]).head.value == "head"
-        and LinkedList([3, "three"]).head.next.value == 3
-    )
+    assert LinkedList([3, "three"]).head.value == "three"
+    assert LinkedList([3, "three"]).head.next.value == 3
```
Comment (W-Ely) [inline tests/code_fellows/test_linked_list.py:10]: Be sure to include a test that ensure an empty LL can be created.  `ll = LinkedList()`

File: tests/code_fellows/test_linked_list.py [hunk 5]
```diff
@@ -1,21 +1,45 @@
...
 
 
 def test_Linked_List_push():
     ll = LinkedList([3, "three"])
     assert ll.head.value == "three"
     assert ll.head.next.value == 3
-    ll.push("new_head")
+    ll.push(["new_head"])
```
Comment (W-Ely) [inline tests/code_fellows/test_linked_list.py]: One pushes a single value to a linked list, so the `ll.push("new_head")` was corrrect, but the code is wrong.  Can comment on it since one can only add comments here on code that has changed.  Another good reason to never merge code before it has been reviewed.

File: tests/code_fellows/test_linked_list.py [hunk 6]
```diff
@@ -1,21 +1,45 @@
...
+
+
+def test_Linked_List_search():
+    ll = LinkedList([3, "three", "nine"])
+    assert ll.search(3).value == 3
+    expected = ll.head.next.next
+    assert ll.search(3) == expected
+
+
+def test_linked_list_pop():
```
Comment (W-Ely) [inline tests/code_fellows/test_linked_list.py]: Due to your current implementation the size() will be correct after this action, but nevertheless anything that changes should be reflected in the tests. So after a push or pop, the size needs to be tested too.

**Review by W-Ely — CHANGES_REQUESTED** [review]

File: tests/code_fellows/test_linked_list.py (hunk 5 above)
Comment (nxnode) [reply tests/code_fellows/test_linked_list.py]: ll.push("new_head") end up acting like this:
ll.push("n")
ll.push("e")
ll.push("w")
ll.push("_")
ll.push("h")
ll.push("e")
ll.push("a")
ll.push("d")

So that ll.head.value == 'd'

I see that is due to:
```
    def push(self, values):
        for value in values:
            self.head = Node(value, next=self.head)
```
I was trying to be future oriented with the for loop - thinking about throwing several values at it at once. But I see that is going to require a bit more nuance than just the simple for loop I made. 

I changed the push function to just do what was asked (haha - you know, the actual assignment ;-) but I am still going to keep it in the follow-up because after all of the initial functions are testing properly, it would be beneficial to modify `push()` to handle something like `push("this", 1, "that")` or `ll.push(["new_head", 3, "test"])` and have the function not return ll.head.value as `["new_head", 3, "test"]` or any of the other odd things you could throw at it that might act differently than expected.

File: tests/code_fellows/test_linked_list.py (hunk 6 above)
Comment (nxnode) [reply tests/code_fellows/test_linked_list.py]: That's tricky! I kind of was doing something similar by adding asserts for size() inside of the pop() and push() tests BUT then I remembered I need to isolate everything as much as possible to ensure my tests are actually testing the individual function. Because really, what is the difference between adding an assert in pop() and/or push() to test for size() vs adding an assert in size() to test for pop() and/or push()?

Or should we view classes differently in that regard since all of the functions are part of that one class? Which I guess makes sense. So, in that sense, would it be better suited to test the class as a whole and have a test that just asserts a flow of the functions together? Am I overthinking it?

Comment (W-Ely) [reply tests/code_fellows/test_linked_list.py]: Over thinking it.  `size` should be tested more or less by itself, but anything that could effect the abstract concept of size should include a test for size in some fashion.

One approach is to have a test like
`def test_linked_list_pop_correct_size_is_maintained():`
